features/net/FEATURE_IPV4/lwip-interface/lwipopts.h

```diff
@@ -66,6 +66,7 @@
 
 #define LWIP_DHCP                   1
 #define LWIP_DNS                    1
+#define LWIP_SOCKET                 0
```

Comment:
It disables the BSD-compatible socket API that lwip provides, which is unused in the network-socket API. The only change would be a small decrease in code size.